sm-shop/src/main/java/com/salesmanager/shop/store/controller/customer/facade/CustomerFacadeImpl.java

```diff
@@ -1071,4 +1056,68 @@ public void changePassword(Customer customer, String newPassword) {
     }
     
   }
+
+    @Override
+    public ReadableCustomer create(PersistableCustomer customer, MerchantStore store, Language language, MultipartFile profilePhoto) {
+      Validate.notNull(customer, "Customer cannot be null");
+      Validate.notNull(customer.getEmailAddress(), "Customer email address is required");
+
+      //set customer user name
+      customer.setUserName(customer.getEmailAddress());
+      if (userExist(customer.getUserName())) {
+        throw new ServiceRuntimeException("User already exist");
+      }
+      //end user exists
+
+      Customer customerToPopulate = convertPersistableCustomerToCustomer(customer, store);
+      try {
+        setCustomerModelDefaultProperties(customerToPopulate, store);
+      } catch (Exception e) {
+        throw new ServiceRuntimeException("Cannot set default customer properties",e);
+      }
+      saveCustomer(customerToPopulate);
+      customer.setId(customerToPopulate.getId());
+
+      notifyNewCustomer(customer, store, customerToPopulate.getDefaultLanguage());
+      //convert to readable
+      return convertCustomerToReadableCustomer(customerToPopulate, store, language);
+    }
+
+    private PersistableCustomer update(PersistableCustomer customer, MerchantStore store, boolean generateRandomPassword) {
+      if (customer.getId() == null || customer.getId() == 0) {
+        throw new ServiceRuntimeException("Can't update a customer with null id");
+      }
+
+      Customer customerModel = customerService.getById(customer.getId());
+
+      try{
+        customerPopulator.populate(customer, customerModel, store, store.getDefaultLanguage());
+      } catch (ConversionException e) {
+        throw new ConversionRuntimeException(e);
+      }
+
+      String password = customer.getPassword();
+      if (StringUtils.isBlank(password) && generateRandomPassword) {
+        password = UserReset.generateRandomString();
+        customer.setPassword(password);
+      }
+
+      Optional.ofNullable(customer.getProfilePhoto()).ifPresent(profilePhotoFile -> {
```

Comment:
isn't it simpler to use if null?

Reply:
Maybe you have a point here but let's have a concept of using the new features of java and that was my recommendation